WordPress/gutenberg
# Post Template: Output number of columns as classname for backwards compatibility

## What?


Fixes #51346 

Output `columns-x` classname for the post template block when the grid layout is in use and when a `columnCount` value is set (where `x` is the number of columns).

## Why?


As raised in #51346, this is so that themes that are currently targeting post template blocks with a certain number of columns can continue to do so.

This change is only for backwards compatibility / to ensure themes can continue to target these blocks, as the old width-based columns class styling rules (e.g. [here](https://github.com/WordPress/gutenberg/blob/555749a8990314f5f2e122b6957623d5b9844c8e/packages/block-library/src/post-template/style.scss#L27)) no longer apply for grid layouts.

## How?


* Add logic to the post template block's PHP and JS output at render time that includes a `columns-${ columnCount }` classname.
* The classname is only output when the layout type is `grid` because the columnCount number isn't automatically removed when switching between grid and flow layouts in the editor. Personally, I think this is desired behaviour, as it means that someone switching between values doesn't lose their old value, but it does mean we need to check for `grid` before outputting the classname (which I think is fine).

## Testing Instructions





1. Add a Query Loop block to a post, page, or template
2. Set its Post Template block to Grid
3. View the source in the editor, and note that you should see the `columns-3` classname or whichever number of columns you have set
4. Save and load the site frontend and double check that the classname is output as expected
5. Back in the editor, switch the Post Template block to flow layout and inspect the markup. The `columns-3` classname should no longer appear.
6. Save and then reload the site frontend. The `columns-3` classname should no longer appear there either.

## Screenshots or screencast 

| Editor markup | Site frontend markup|
| --- | --- |
| <img width="566" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/98bb8725-3465-4114-819b-8a88d5682495"> | <img width="804" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/f2ab4c92-abe5-4ba1-b828-222dc3bb4bb8"> |
